src/yunohost/data_migrations/0009_dh_params.py

```diff
@@ -0,0 +1,36 @@
+import re
+import os
+
+from moulinette.utils.log import getActionLogger
+from moulinette.utils.filesystem import chown
+
+from yunohost.tools import Migration
+from yunohost.service import service_regen_conf
+
+command = "nice -n 19 openssl dhparam -out /etc/ssl/private/dh2048.pem -outform PEM -2 2048 -dsaparam 2> /var/log/yunohost/dhparam_generation.log && chown root:ssl-cert /etc/ssl/private/dh2048.pem && rm /etc/cron.hourly/yunohost-generate-dh-params\n"
+dhparams_file = "/etc/ssl/private/dh2048.pem"
+
+class MyMigration(Migration):
+    "This migration will add dh_params line and generate it in installed instance"
+
+    def migrate(self):
+
+        try:
+            file_open = open(dhparams_file)
```

Comment:
Ah ... Now I realize what you are doing here with this try/except ...

Personally I would rather just have a 

```
if os.path.exists(dhparams_file):
    return
```

(and just assume that if the file exist, the admin edited its nginx conf file appropriately already) instead of those "pythonic" try/except where the intention is implicit :/

Reply:
You are true. I just didn't know the function :/ (shame on me). Well I tried to find a solution. :D

Reply:
done.